ark/utils/io_utils_test.py

```diff
@@ -184,13 +176,48 @@ def test_list_folders():
         pathlib.Path(os.path.join(temp_dir, 'test_badfile.txt')).touch()
 
         # test substrs is None (default)
-        get_all = iou.list_folders(temp_dir)
-        assert get_all.sort() == dirnames.sort()
+        get_all = iou.list_folders(temp_dir, exact_match=False)
+        assert sorted(get_all) == sorted(dirnames)
 
         # test substrs is not list (single string)
-        get_txt = iou.list_folders(temp_dir, substrs='_txt')
-        assert get_txt.sort() == dirnames[0:2].sort()
+        get_txt = iou.list_folders(temp_dir, substrs='_txt', exact_match=False)
+        assert sorted(get_txt) == sorted(dirnames[0:2])
 
         # test substrs is list
-        get_test_and_other = iou.list_folders(temp_dir, substrs=['test_', 'other'])
-        assert get_test_and_other.sort() == dirnames[1:].sort()
+        get_test_and_other = iou.list_folders(
+            temp_dir, substrs=['test_', 'other'], exact_match=False
+        )
+        assert sorted(get_test_and_other) == sorted(dirnames[1:])
+
+    # Tests "Exact Substring Matching", `exact_match` = True
+    with tempfile.TemporaryDirectory() as temp_dir:
```

Comment:
Why do we need to re-generate the context with the temp_dir. Can't we use the same files as above?